nuxeo-features/nuxeo-annotation/nuxeo-annotation-api/src/main/java/org/nuxeo/ecm/annotation/Annotation.java

```diff
@@ -29,52 +29,201 @@
  */
 public interface Annotation {
 
+    /**
+     * Gets annotation id.
+     * 
+     * @return the id
+     */
+    String getId();
+
+    /**
+     * Sets annotation id.
+     * 
+     * @param id the id
+     */
+    void setId(String id);
+
+    /**
+     * Gets annotation color.
+     * 
+     * @return the color, expressed in hexadecimal
+     */
     String getColor();
 
+    /**
+     * Sets annotation color.
+     * 
+     * @param color the color, expressed in hexadecimal
+     */
     void setColor(String color);
 
+    /**
+     * Gets annotation modification date.
+     * 
+     * @return the modification date
+     */
     Calendar getDate();
 
+    /**
+     * Sets annotation modification date.
+     * 
+     * @param date the modification date
+     */
     void setDate(Calendar date);
 
+    /**
+     * Gets annotation list of flags, separated by commas. Possible values are:
+     * invisible|hidden|print|nozoom|norotate|noview|readonly|locked|togglenoview
+     *
+     * @return the flags
+     */
     String getFlags();
 
+    /**
+     * Sets annotation list of flags.
+     *
+     * @param flags the flags, separated by commas
+     */
     void setFlags(String flags);
 
+    /**
+     * Gets annotation name.
+     * 
+     * @return the name
+     */
     String getName();
 
+    /**
+     * Sets annotation name.
+     * 
+     * @param name the name
+     */
     void setName(String name);
 
+    /**
+     * Gets the annotated document id.
+     * 
+     * @return the annotated document id
+     */
     String getDocumentId();
 
+    /**
+     * Sets the annotated document id.
+     *
+     * @param documentId the annotated document id
+     */
     void setDocumentId(String documentId);
 
+    /**
+     * Gets the annotated document xpath.
+     *
+     * @return the annotated document xpath
+     */
+    String getDocumentXpath();
+
+    /**
+     * Sets the annotated document xpath.
```

Comment:
Sets the xpath of annotated blob in the document. Call it `setXpath` too, it's the blob xpath, not the document xpath.